test/service/retry.handler.service.spec.ts

```diff
@@ -46,106 +57,224 @@ describe(`OE Unauthorised Response Handler test suite`, () => {
+      expect(mockDebugRequestLog).not.toHaveBeenCalled();
     });
 
     test(`${fnNamePutOE} should retry after unauthorised response`, async () => {
       const mockResponse = { httpStatusCode: 401 };
       mockPutOverseasEntity.mockResolvedValueOnce( mockResponse);
 
-      await makeOverseasEntitiesApiCallWithRetry(fnNamePutOE, req, session, ...otherParamsPutOE);
+      await makeApiCallWithRetry(serviceNameOE, fnNamePutOE, req, session, ...otherParamsPutOE);
 
       expect(mockPutOverseasEntity).toBeCalledWith(TRANSACTION_ID, OVERSEAS_ENTITY_ID, APPLICATION_DATA_MOCK);
       expect(mockPutOverseasEntity).toHaveBeenCalledTimes(2);
       expect(mockRefreshToken).toHaveBeenCalledTimes(1);
       expect(mockCreateApiClient).toHaveBeenCalledTimes(2);
 
-      const responseMsg = `Retrying ${fnNamePutOE} call after unauthorised response`;
+      const responseMsg = `Retrying ${fnNamePutOE} call on ${serviceNameOE} service after unauthorised response`;
       expect(mockDebugRequestLog).toBeCalledWith(req, `${responseMsg} - ${JSON.stringify(mockResponse)}`);
-      expect(mockDebugRequestLog).toHaveBeenCalledTimes(2);
+      expect(mockDebugRequestLog).toHaveBeenCalledTimes(1);
 
+      expect(mockInfoRequestLog).toHaveBeenCalledTimes(2);
+      expect(mockInfoRequestLog).toBeCalledWith(req, `Making a ${fnNamePutOE} call on ${serviceNameOE} service with token accessToken`);
       expect(mockInfoRequestLog).toBeCalledWith(req, `New access token: ${mockNewAccessToken}`);
-      expect(mockInfoRequestLog).toHaveBeenCalledTimes(1);
     });
 
     test(`${fnNamePutOE} should return immediately the error if not 401`, async () => {
       const mockResponse = { httpStatusCode: 500 };
       mockPutOverseasEntity.mockResolvedValueOnce( mockResponse);
 
-      const response = await makeOverseasEntitiesApiCallWithRetry(fnNamePutOE, req, session, ...otherParamsPutOE);
+      const response = await makeApiCallWithRetry(serviceNameOE, fnNamePutOE, req, session, ...otherParamsPutOE);
 
       expect(mockPutOverseasEntity).toBeCalledWith(TRANSACTION_ID, OVERSEAS_ENTITY_ID, APPLICATION_DATA_MOCK);
       expect(response).toEqual(mockResponse);
 
       expect(mockPutOverseasEntity).toHaveBeenCalledTimes(1);
       expect(mockCreateApiClient).toHaveBeenCalledTimes(1);
+      expect(mockInfoRequestLog).toHaveBeenCalledTimes(1);
+      expect(mockDebugRequestLog).not.toHaveBeenCalled();
+    });
+
+  });
+
+});
+
+const serviceNameTransaction = "transaction";
+const fnNamePostTransaction = "postTransaction";
+const fnNamePutTransaction = "putTransaction";
+
+describe(`Transaction Unauthorised Response Handler test suite`, () => {
+
+  beforeEach (() => {
+    jest.clearAllMocks();
+  });
+
+  describe(`${serviceNameTransaction}.${fnNamePostTransaction} calls`, () => {
+
+    test(`${fnNamePostTransaction} should responde with created OE resource`, async () => {
+      const mockResponse = { httpStatusCode: 201, resource: { id: OVERSEAS_ENTITY_ID } };
```

Comment:
Looks odd that OVERSEAS_ENTITY_ID is mentioned here. Shouldn't this be a TRANSACTION_ID?

Reply:
Good spot, I will fix it